Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -556,6 +601,19 @@ TEST_F(SerializerTest, BadNullValue) {
       Status(FirestoreErrorCode::DataLoss, "ignored"), bytes);
 }
 
+/* TODO(rsgowman): If the number (representing a bool) on the wire is not 1 or
+ * 0, then nanopb happily drops the value into the bool anyways, but does so via
+ * an integer pointer. Something like this:
+ *   bool b;
+ *   *(int*)b = 2;
+ * Depending on the compiler, the resulting bool could be *both* true *and*
+ * false. eg at least gcc does this. (Effectively, comparing to true is
+ * implemented via != 0, and comparing to false is implemented via != 1.)
+ *
+ * We need to figure out what, if anything, we want to do about this. For now,
+ * we'll just ignore this test.
+ */
+#if 0
 TEST_F(SerializerTest, BadBoolValue) {
```

Comment:
I think you can just do TEST_F(SerializerTest, DISABLED_BadBoolValue) instead of the condition. But I could be wrong due to our customized setup of the C++ test in Xcode project.

Reply:
Hm, interesting. I'll leave it as is for now; that's guaranteed to work everywhere. (I suspect the eventual answer might be to just delete the test.)